### PR TITLE
A11Y: Add keyboard support for do-not-disturb modal

### PR DESCRIPTION
**Pull request by pmusaraj**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/tap-tile.js [hunk 1]
```diff
@@ -5,25 +5,25 @@ import { propertyEqual } from "discourse/lib/computed";
...
   tabIndex: 0,
-
   ariaPressed: reads("active"),
 
   click() {
     this.onChange(this.tileId);
   },
 
+  keyDown(e) {
+    if (e.which === 13) {
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/components/tap-tile.js:23]: Minor suggestion but I think assigning the number to a constant can help to improve readability.

File: app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js [hunk 2]
```diff
@@ -48,6 +48,33 @@ acceptance("Do not disturb", function (needs) {
...
+    await visit("/");
+    await click(".header-dropdown-toggle.current-user");
+    await click(".menu-links-row .user-preferences-link");
+
+    await click(".do-not-disturb");
+
+    assert.ok(exists(".do-not-disturb-modal"), "modal to choose time appears");
+
+    let tiles = queryAll(".do-not-disturb-tile");
+    assert.ok(tiles.length === 4, "There are 4 duration choices");
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js:63]:
```suggestion
    assert.strictEqual(tiles.length, 4, "There are 4 duration choices");
```

The diff will be more useful when the assertion fails.

Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js:63]:
```suggestion
    assert.striclEqual(queryAll(".do-not-disturb-tile").length, 4, "There are 4 duration choices");
```

Comment (tgxworld) [reply app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js:63]: Removes the unneeded variable and also makes the diff output more useful should the assertion fails

File: app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js [hunk 3]
```diff
@@ -48,6 +48,33 @@ acceptance("Do not disturb", function (needs) {
...
+
+    await triggerKeyEvent(".do-not-disturb-tile:nth-child(1)", "keydown", 13);
+
+    assert.ok(
+      query(".do-not-disturb-modal").style.display === "none",
+      "modal is hidden"
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js:69]:
```suggestion
    assert.strictEqual(
      query(".do-not-disturb-modal").style.display,
      "none",
      "modal is hidden"
```

File: app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js [hunk 4]
```diff
@@ -48,6 +48,33 @@ acceptance("Do not disturb", function (needs) {
...
+    );
+
+    assert.ok(
+      exists(".header-dropdown-toggle .do-not-disturb-background"),
+      "moon icon is present in header"
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js:74]: I think the word is an internal implementation detail and the query above does not reflect that as well. Should the query be more specific in this case?

File: app/assets/javascripts/discourse/app/components/tap-tile.js (hunk 1 above)
Comment (pmusaraj) [reply app/assets/javascripts/discourse/app/components/tap-tile.js:23]: Sure, we can actually do 
```suggestion
    if (e.key === "Enter") {
```

**Review by Chapoi — APPROVED** [review]

**Pull request merged**